### PR TITLE
Implement Secret Key write object

### PR DESCRIPTION
**Pull request by metsma**
Signed-off-by: Raul Metsma <raul@metsma.ee>

##### Checklist

- [ ] Documentation is added or updated
- [ ] New files have a LGPL 2.1 license statement
- [X] PKCS#11 module is tested
- [ ] Windows minidriver is tested
- [ ] macOS tokend is tested

### CODE REVIEW

File: src/tools/pkcs11-tool.c [hunk 1]
```diff
@@ -3153,9 +3153,63 @@ static int write_object(CK_SESSION_HANDLE session)
...
-	if (opt_object_class == CKO_DATA) {
+		break;
+	case CKO_SECRET_KEY:
+		clazz = CKO_SECRET_KEY;
+		type = CKK_AES;
+
+		if (opt_key_type != 0) {
+			if (strncmp(opt_key_type, "AES:", strlen("AES:")) == 0 || strncmp(opt_key_type, "aes:", strlen("aes:")) == 0)
+				type = CKK_AES;
+			else if (strncmp(opt_key_type, "DES:", strlen("DES:")) == 0 || strncmp(opt_key_type, "des:", strlen("des:")) == 0)
```
Comment (martinpaljak) [inline src/tools/pkcs11-tool.c]: Not sure if any new code should be added for DES (as in single key DES) whereas there are plenty of systems that make use of 2 key 3DES (and would be nice to have explicit support for this in import level, without having to do manual key shuffling before)

Comment (metsma) [reply src/tools/pkcs11-tool.c]: I think this option list should to be equal whith generate key options.
https://github.com/OpenSC/OpenSC/blob/master/src/tools/pkcs11-tool.c#L2460
Or will I remove this option from gen_key also?

File: src/tools/pkcs11-tool.c [hunk 2]
```diff
@@ -3153,9 +3153,61 @@ static int write_object(CK_SESSION_HANDLE session)
...
-	if (opt_object_class == CKO_DATA) {
+		break;
+	case CKO_SECRET_KEY:
+		clazz = CKO_SECRET_KEY;
+		type = CKK_AES;
+
+		if (opt_key_type != 0) {
+			if (strncmp(opt_key_type, "AES:", strlen("AES:")) == 0 || strncmp(opt_key_type, "aes:", strlen("aes:")) == 0)
+				type = CKK_AES;
+			else if (strncmp(opt_key_type, "DES3:", strlen("DES3:")) == 0 || strncmp(opt_key_type, "des3:", strlen("des3:")) == 0)
```
Comment (frankmorgner) [inline src/tools/pkcs11-tool.c]: `strncasecmp()`?

File: src/tools/pkcs11-tool.c [hunk 3]
```diff
@@ -3153,9 +3153,61 @@ static int write_object(CK_SESSION_HANDLE session)
...
+		if (opt_object_label != NULL) {
+			FILL_ATTR(seckey_templ[n_seckey_attr], CKA_LABEL, opt_object_label, strlen(opt_object_label));
+			n_seckey_attr++;
+		}
+		if (opt_object_id_len != 0)  {
+			FILL_ATTR(seckey_templ[n_seckey_attr], CKA_ID, opt_object_id, opt_object_id_len);
+			n_seckey_attr++;
+		}
+
+		rv = p11->C_CreateObject(session, seckey_templ, n_seckey_attr, &seckey_obj);
```
Comment (frankmorgner) [inline src/tools/pkcs11-tool.c]: for consistency, please move this part down to where the other objects are created.

**Pull request merged**